### PR TITLE
expression: fix expression that duration type null equal with const null

### LINKED ISSUE #56744: time type query `<=> null`  failed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
create table lrr(`COL1` time DEFAULT NULL,`COL2` time DEFAULT NULL);
insert into lrr(col2) values("-229:53:34");
select * from lrr where col1 <=> null;
```

### 2. What did you expect to see? (Required)
+------+------------+
| COL1 | COL2       |
+------+------------+
| NULL | -229:53:34 |
+------+------------+

### 3. What did you see instead (Required)
Empty set

### 4. What is your TiDB version? (Required)

Release Version: v8.4.0
Edition: Community
Git Commit Hash: 2205f332ffcf3553ffc49438cfe1087a0f08dd18
Git Branch: HEAD
UTC Build Time: 2024-10-17 07:36:44
GoVersion: go1.23.2
Race Enabled: false
Check Table Before Drop: false
Store: tikv

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: close #56744

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix issue that fail to evaluate the expression that duration type null equal with const null
```

### CODE REVIEW

File: pkg/expression/builtin_compare.go [hunk 1]
```diff
@@ -1634,6 +1634,10 @@ func (c *compareFunctionClass) handleDurationTypeComparison(ctx BuildContext, ar
 	if arg0IsCon && arg0Const.DeferredExpr == nil && !arg1IsCon && arg1.GetType(ctx.GetEvalCtx()).GetType() == mysql.TypeDuration {
 		isNull, err = castToDurationIsNull(ctx, arg0)
```
Comment (XuHuaiyu) [inline pkg/expression/builtin_compare.go:1635]:
``` go
if arg0Const.Value.IsNull()
```

Comment (xzhangxian1008) [reply pkg/expression/builtin_compare.go:1639]: > ```go
> if arg0Const.Value.IsNull()
> ```

fixed

**Review by XuHuaiyu — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**